### PR TITLE
Feature/setup

### PR DESCRIPTION
**Pull request by MeqdamAlqudah**
1. Create class Person with the following:
- Instance vars: @id, @name, and @age.
- Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
- Getters for @id, @name, and @age.
- Setters for @name and @age.
- Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
- Public method can_use_services? that returns true if person is of age or if they have permission from parents.

2. Create class Student with the following:

- Inherits from Person.
- The constructor extends the parent's constructor by adding @classroom and a parameter for it.
- Method play_hooky that returns "¯\(ツ)/¯".

3. Create a class Teacher with the following:

- Inherits from Person.
- The constructor extends the parent's constructor by adding @specialization and a parameter for it.
- Override can_use_services? so it always returns true.

### CODE REVIEW

File: school_library_people.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    @id = 1
```
Comment (mmsesay) [inline school_library_people.rb]: - [OPTIONAL] You did well adding this id. What do you think about making it dynamically populated?

`@id = Random.rand(1..1000)`

File: school_library_people.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  # rubocop:disable Naming/PredicateName
+  def is_of_age?
+    @age > 18
+  end
+  # rubocop:enable Naming/PredicateName
+  private :is_of_age?
+  def can_use_services?
+    is_of_age? || @parent_permission ? true : false
```
Comment (mmsesay) [inline school_library_people.rb]: - [OPTIONAL] Ruby is clean in terms of syntax. You can simply achieve the same but just wiring `is_of_age?() || @parent_permission`

File: school_library_people.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    @parent_permission = parent_permission
+    @id = 1
+  end
+
+  # rubocop:disable Naming/PredicateName
+  def is_of_age?
+    @age > 18
+  end
+  # rubocop:enable Naming/PredicateName
+  private :is_of_age?
```
Comment (mmsesay) [inline school_library_people.rb]: - [OPTIONAL] It is a good practice to set the private keyword on its own line and all other functions below it are automatically taken to be private methods.

See example:
```ruby
private

def testone; end
def testtwo; end

```


From the example, testone and testtwo are private methods.

File: school_library_people.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    @id = 1
+  end
+
+  # rubocop:disable Naming/PredicateName
+  def is_of_age?
+    @age > 18
```
Comment (mmsesay) [inline school_library_people.rb]: - [ ] This check is to verify if the age is greater or equal to 18. What you checked for was if the age is greater than 18.

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @MeqdamAlqudah, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada:  :champagne:  :confetti_ball: 
Hi  :wave:   :wave: @MeqdamAlqudah , 

200w 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 
- Your README file looks very professional :+1: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**